src/Lister/SimpleSubPageFinder.php

```diff
@@ -201,6 +219,11 @@ private function getOptions() {
 
 		$options['LIMIT'] = (int)$this->options[self::OPT_LIMIT];
 
+		$sort_oder = $this->options[self::OPT_SORT_ORDER];
+		if ($sort_oder != self::NONE) {
+            $options['ORDER BY'] = "LENGTH(page_title) $sort_oder, page_title $sort_oder";
```

Comment:
This is naive "native sort". It is not so polished as PHP's `strnatcmp` but works fine in the most cases.